testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/ElementAssertj.java

```diff
@@ -229,6 +230,65 @@ public void hasActiveElementsExactly(
         });
   }
 
+  public void hasCompletedElementsInOrder(
+      final long processInstanceKey, final List<ElementSelector> elementSelectors) {
+
+    awaitFlowNodeInstanceAssertion(
+        processInstanceFilter(processInstanceKey),
```

Comment:
:wrench: We can filter the flow-node instances by their state on the query here instead of filtering the query result afterward.